client/infinispan-key-value-store-hotrod/src/test/java/org/infinispan/api/client/impl/KeyValueStoreTest.java

```diff
@@ -0,0 +1,145 @@
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.test.HotRodClientTestingUtil;
+import org.infinispan.client.hotrod.test.SingleHotRodServerTest;
+import org.infinispan.functional.FunctionalTestUtils;
+import org.infinispan.server.core.admin.embeddedserver.EmbeddedServerAdminOperationHandler;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.hotrod.configuration.HotRodServerConfigurationBuilder;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import io.reactivex.Flowable;
+import io.reactivex.subscribers.TestSubscriber;
+
+@Test(groups = "functional", testName = "org.infinispan.api.client.impl.KeyyValueStoreSimpleTest")
+public class KeyValueStoreTest extends SingleHotRodServerTest {
+
+   public static final String CACHE_NAME = "test";
+   private Infinispan infinispan;
+
+   private KeyValueStore<Integer, String> store;
+
+   @Override
+   protected HotRodServer createHotRodServer() {
+      HotRodServerConfigurationBuilder serverBuilder = new HotRodServerConfigurationBuilder();
+      serverBuilder.adminOperationsHandler(new EmbeddedServerAdminOperationHandler());
+      cacheManager.administration().createCache(CACHE_NAME, new org.infinispan.configuration.cache.ConfigurationBuilder().build());
+      return HotRodClientTestingUtil.startHotRodServer(cacheManager, serverBuilder);
+   }
+
+   @Override
+   protected void setup() throws Exception {
+      super.setup();
+      /**
+       * In real world example, we should only need to call this way
+       * Infinispan infinispan = InfinispanClient.newInfinispan();
+       */
+      infinispan = new InfinispanClientImpl(remoteCacheManager);
+      store = FunctionalTestUtils.await(infinispan.getKeyValueStore(CACHE_NAME, KeyValueStoreConfig.defaultConfig()));
+   }
+
+   @Override
+   protected void teardown() {
+      await(infinispan.stop());
+      super.teardown();
+   }
+
+   @BeforeMethod
+   public void clearStoreBeforeEachTest() {
+      KeyValueStore<Integer, String> store = FunctionalTestUtils.await(infinispan.getKeyValueStore(CACHE_NAME, KeyValueStoreConfig.defaultConfig()));
+      await(store.clear());
+   }
+
+   public void testConfiguration() {
+      // TODO: new configuration API https://issues.jboss.org/browse/ISPN-9929
+      ClientConfig clientConfig = new ClientConfigurationLoader.ConfigurationWrapper(new ConfigurationBuilder().build());
+      Infinispan client = InfinispanClient.newInfinispan(clientConfig);
+      assertNotNull(client);
+      await(client.stop());
+   }
+
+   public void testGetNoValue() {
+      assertNull(await(store.get(0)));
+   }
+
+   public void testCreate() {
+      await(store.insert(1, "hi"));
+      assertEquals("hi", await(store.get(1)));
+   }
+
+   public void testSaveMany() {
+      await(store.saveMany(Flowable.fromArray(new AbstractMap.SimpleEntry(1, "adios"),
```

Comment:
the `saveMany()` seems to be broken.
this test fails:
```
   public void testSaveMany2() {
      Map<Integer, String> map = new HashMap<>();
      for (int i = 0; i < 100; ++i) {
         map.put(i, UUID.randomUUID().toString());
      }
      await(store.saveMany(Flowable.fromIterable(map.entrySet())));

      for (int i = 99; i >= 0; --i) {
         assertEquals(map.get(i), await(store.get(i)));
      }
   }
```

Reply:
it's broken for what you pointed before with the CompletionStage
test passes sometimes and does not other times ...
but I'm reconsidering not only implementation but the signature